### ISSUE TITLE
Ubuntu install without docker fails

### ISSUE THREAD

**Issue by pwhipp** [golemfactory/clay#1222]
Following the instructions in https://github.com/golemfactory/golem/wiki/Installation, I hit a Docker dependency problem when calling `python setup.py develop`:

```
Using /home/paul/.virtualenvs/golem/lib/python2.7/site-packages
Finished processing dependencies for golem==0.7.0
Traceback (most recent call last):
  File "setup.py", line 100, in <module>
    DockerManager.pull_images()
  File "/home/paul/wk/blockchain/golem/golem/docker/manager.py", line 344, in pull_images
    if not cls.command('images', args=[version], check_output=True):
  File "/home/paul/wk/blockchain/golem/golem/docker/manager.py", line 296, in command
    stdin=DEVNULL)
  File "/usr/lib/python2.7/subprocess.py", line 567, in check_output
    process = Popen(stdout=PIPE, *popenargs, **kwargs)
  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
```

**Comment by screaminbug** [comment]
Same thing on Windows

```
Using c:\python27\lib\site-packages
Finished processing dependencies for golem==0.7.1
Traceback (most recent call last):
  File "setup.py", line 100, in <module>
    DockerManager.pull_images()
  File "C:\Users\Tomislav\PycharmProjects\golem\golem\docker\manager.py", line 344, in pull_images
    if not cls.command('images', args=[version], check_output=True):
  File "C:\Users\Tomislav\PycharmProjects\golem\golem\docker\manager.py", line 296, in command
    stdin=DEVNULL)
  File "C:\Python27\lib\subprocess.py", line 212, in check_output
    process = Popen(stdout=PIPE, *popenargs, **kwargs)
  File "C:\Python27\lib\subprocess.py", line 390, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 640, in _execute_child
    startupinfo)
WindowsError: [Error 2] The system cannot find the file specified
```

**Comment by mfranciszkiewicz** [comment]
Check whether you:
- can successfuly run `docker images` in terminal
  **Windows**: run `docker-machine start golem` and follow the instructions from `docker-machine env golem` first. If `docker` or `docker-machine` cannot be run, please add the docker installation directory to your `PATH` variable.
- have sufficient privileges
  **Linux**: add user to group `docker` or ~run Golem with sudo~

In case of further problems, please refer to https://docs.docker.com/toolbox/faqs/troubleshoot/

**Comment by pwhipp** [comment]
I was not setting up with docker. As I'm running Linux and want to play with the code without the docker fluff I was attempting the 'native' installation detailed under the "Ubuntu: Alternative method using virtualenv" section in the installation guide.

**Comment by mfranciszkiewicz** [comment]
The `virtualenv` section describes an alternative way of managing and installing python dependencies. Docker is still required.

**Comment by pwhipp** [comment]
Installing docker did not address the problem so I tried using the install.sh.

While it ran, I got "E: Package 'docker-engine' has no installation candidate" under the HyperG install section but it continued. No other errors were reported.

With docker installed, and the full install run (albeit with the above reported error), I still get exactly the same error when I try to run python setup.

My system already has geth/ethereumwallet etc. installed and working. docker is installed and running.
